lib/fizzy/parser_expr.cpp

```diff
@@ -408,8 +411,12 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, bool have
         }
         }
         code.instructions.emplace_back(instr);
+        max_stack_depth += instr_stack_change(instr);
```

Comment:
I think we need to move this into/after a validation phase so that we have access to each of the sections (such as types) to know how many stack items are we popping/pushing (for example in `call`/`call_indirect`/etc).